### PR TITLE
#421 Inviting user by role and email

### PR DESCRIPTION
**Pull request by niyobertin**
### **PR Description**
This PR involves sending invtiation to the user to register to any organization by provindign emai and role.

### **Description of tasks that were expected to be completed**
implementing the model to that will be opened when a user click on invite button
Adding intergaration to send an invitation

### **How has this been tested?**
To test this: Clone the this repostory, cd into it then run the following commands:

npm install : to install dependencies
npm run dev : to sart the server
After that navigate to l[ocahost](http://localhost:3001/) anvigate to the invition page. then click on invite button the model wibb open. Enter email and select the role. the send you will se the toast notification that invitation sent.

### **Number of Commits**
2 commits

### Screeshoot 

**Desktop** 
Screenshot 2024-09-04 at 16 28 23

Screenshot 2024-09-04 at 16 28 05

**Mobile**
Screenshot 2024-09-06 at 09 41 41
Screenshot 2024-09-06 at 09 42 12

### **Please check this Checklist before you submit your PR:**

- [x]  My code follows the style guidelines of this project
- [x]  I have performed a self-review of my code
- [x]  I have commented my code, particularly in hard-to-understand areas
- [x]  My code generate no warnings
- [x]  My test coverage meet the set test coverage threshold
- [x]  There are no vulnerabilities
- [x]  There are no conflicts with the base branch

### CODE REVIEW

**Review by YvetteNyibuka — APPROVED** [review]

**Review by YvetteNyibuka — APPROVED** [review]

**Review by jniyonkuru — APPROVED** [review]

**Review by Sophie20202 — APPROVED** [review]

**Review by Sophie20202 — APPROVED** [review]

File: src/components/DataPagination.tsx
Comment (ncutixavier) [inline src/components/DataPagination.tsx:1]: Unnecessary changes

File: src/components/invitationModel.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+const roles = ['trainee', 'admin', 'ttl', 'coordinator'];
+
+function InviteForm({ onClose }: { onClose: () => void }) {
+  const [email, setEmail] = useState('');
+  const [role, setRole] = useState('Role');
+  const [orgToken, setOrgToken] = useState('');
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [emailError, setEmailError] = useState('');
+  const [sendInvitation, { loading, error }] = useMutation(SEND_INVITATION);
+  const organisationToken = localStorage.getItem('orgToken');
```
Comment (ncutixavier) [inline src/components/invitationModel.tsx:19]: Enforce typescript

File: src/components/invitationModel.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  useEffect(() => {
+    if (organisationToken) {
+      setOrgToken(organisationToken);
+    }
+  }, [organisationToken]);
+
+  const validateEmail = (email: string) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
```
Comment (ncutixavier) [inline src/components/invitationModel.tsx:30]: Validation should in its folder

**Review by ncutixavier — COMMENTED** [review]
Checkout file naming

**Review by ncutixavier — APPROVED** [review]
We need to introduce validations on the frontend
